.travis.yml

```diff
@@ -1,38 +1,44 @@
 sudo: false
 language: python
-python:
-    - '2.6'
-    - '2.7'
-    - '3.3'
-    - '3.4'
-    - '3.5'
-    - '3.6'
-    - pypy
-    - nightly
-
-# command to install dependencies
-install: "pip install -U tox"
-
-# command to run tests
-env:
-  matrix:
-    - TOXENV=py-pytest28
-    - TOXENV=py-pytest29
-    - TOXENV=py-pytest30
 
 matrix:
   include:
-    - python: '2.7'
-      env: TOXENV=check
     - python: '3.6'
       env: TOXENV=check
+    - python: '3.6'
```

Comment:
I have to say I've always found this a bit funny/redundant that we're routing through `tox` when Travis is really enabling the same thing via it's own job generation system. This include matrix just ends up being a big combo-list defeating the entire purpose. I think `tox` makes sense if you don't have a generative system built into your CI server already.

That's another thing `tox` doesn't support parallel jobs does it?
That's something that'd be super handy when testing locally.
I'd also be interested in using travis' [stages](https://docs.travis-ci.com/user/build-stages/) to do docs and benchmarking runs in parallel with the regular suite.

Reply:
> his include matrix just ends up being a big combo-list defeating the entire purpose. I think tox makes sense if you don't have a generative system built into your CI server already.

One reason to also use tox in this context is to avoid duplicating logic from it like dependency installation and such.

> I'd also be interested in using travis' stages to do docs and benchmarking runs in parallel with the regular suite.

Actually all jobs already run in parallel, give or take Travis current work availability when the build runs.